rfc/0052-sdk3-full-text-search.md

```diff
@@ -566,7 +566,275 @@ A query that matches nothing.
+cluster.searchQuery("search_index_name", query);
+```
+
+and with the new API:
+
+```java
+SearchRequest request = SearchRequest.searchQuery(SearchQuery.matchAll());
+
+cluster.search("search_index_name", request);
+scope.search("search_index_name", request);
+```
+
+2. A vector search with a single vector query and no traditional FTS query:
+
+```java
+// Using an imaginary API provided by OpenAI to generate the vector.
+float[] vector = OpenAI.createVectorFor("some query");
+
+SearchRequest request = SearchRequest
+        .vectorSearch(VectorSearch.create(VectorQuery.create("vector_field", vector)));
+
+scope.search("search_index_name", request);
+cluster.search("search_index_name", request);
+```
+
+3. A combined search with both vector search and a traditional FTS query:
+
+```java
+float[] vector = OpenAI.createVectorFor("some query");
+
+SearchRequest request = SearchRequest
+        .searchQuery(SearchQuery.matchAll())
+        .vectorSearch(VectorSearch.create(VectorQuery.create("vector_field", vector)));
+
+scope.search("search_index_name", request);
+cluster.search("search_index_name", request);
+```
+
+4. A more complex example showing a vector search containing multiple vector queries, setting all parameters:
+
+```java
+// Sending multiple `VectorQuery`s, and setting all possible parameters.
+SearchRequest request = SearchRequest
+        .vectorSearch(VectorSearch.create(List.of(
+                VectorQuery.create("vector_field", aVector).numCandidates(2).boost(0.3),
+                VectorQuery.create("vector_field", anotherVector).numCandidates(1).boost(0.7)),
+            vectorSearchOptions().vectorQueryCombination(VectorQueryCombination.AND));
+
+scope.search("search_index_name", request);
+cluster.search("search_index_name", request);
+```
+
+### VectorSearch
+`VectorSearch` construction is platform-idiomatic:
+
+`VectorSearch.create(List<VectorQuery> vectorQueries, [VectorSearchOptions options])`
+
+The first parameter needs to be mandatory.
+
+`VectorSearch.create(VectorQuery vectorQuery, [VectorSearchOptions options])` can be added to make it easier to send a single `VectorQuery`.
+This is left optional to better support SDKs without overloads.
+
+`VectorSearchOptions` will currently support just one option:
+
+* `vectorQueryCombination` (`VectorQueryCombination`).  Will be sent in the top-level JSON payload as `knn_operator` (string) as either `"and"` or `"or"`.  We name the field differently in the SDK to a) improve readability and b) be able to better support possible future forms of vector search such as ANN.  If not set by the user, no value is sent.  It controls how elements in the `knn` array are combined.
+
+```
+enum VectorQueryCombination {
+    AND,
+    OR
+}
+```
+
+`VectorSearch` does not extend the `SearchQuery` interface, which is now reserved for traditional FTS queries.
+
+### VectorQuery
+`VectorQuery` creation is platform-idiomatic:
+
+`VectorQuery.create(String vectorFieldName, float[] vector)`
```

Comment:
In order to be more specific and support potential integrations with vector libraries, why not `VectorQuery.fromVector`?

Reply:
It's mentioned that construction is platform-idiomatic and this is intended to include the "VectorQuery.create(...)" naming.  If `new VectorQuery(...)`, or `VectorQuery.fromX(...)`, or some other form, is more idiomatic and better fits the SDK's existing FTS creation methods, it should use such.

Reply:
For SDKs which use them, is this float64 or float32?

Reply:
float32